framework/src/ble_manager/ble_manager_state.c

```diff
@@ -0,0 +1,336 @@
+#include <ble_manager/ble_manager.h>
+#include "ble_manager_event.h"
+#include "ble_manager_msghandler.h"
+#include "ble_manager_state.h"
+#include "ble_manager_log.h"
+
+struct _blemgr_state_handle {
+	blemgr_state_e state;
+	blemgr_server_state server_state;
+	blemgr_server_adv_state adv_state;
+	blemgr_client_state client_state[BLE_MAX_CONNECTION_COUNT];
+	blemgr_client_scan_state scan_state;
+	blemgr_mode_e mode;
+};
+
+typedef struct _blemgr_state_handle blemgr_state_handle_s;
+static blemgr_state_handle_s g_state_handle = {
+	BLEMGR_UNINITIALIZED,
+	BLEMGR_SERVER_NONE,
+	BLEMGR_SERVER_ADV_NONE,
+	{
+		0,
+	},
+	BLEMGR_CLIENT_SCAN_NONE,
+	BLEMGR_UNKNOWN_MODE,
+};
+
+#define BLE_STATE_CHECK                                   \
+	do {                                                  \
+		if (g_state_handle.state != BLEMGR_INITIALIZED) { \
+			BLE_LOG_INFO("[BLEMGR] is not running\n");    \
+			ret = TRBLE_INVALID_STATE;                    \
+			goto handle_req_done;                         \
+		}                                                 \
+	} while (0)
+
+/*
+ * public
+ */
+ble_result blemgr_handle_request(blemgr_msg_s *msg)
+{
+	trble_result_e ret = TRBLE_SUCCESS;
+
+	BLE_LOG_INFO("[BLEMGR] T%d --> _handle_request[%d]\n", getpid(), msg->event);
+
+	switch (msg->event) {
+	case BLE_EVT_INIT_CMD: {
+		if (g_state_handle.state != BLEMGR_UNINITIALIZED) {
+			BLE_LOG_INFO("[BLEMGR] already running[mode : %d]\n", g_state_handle.mode);
+			break;
+		}
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_client_init_config *client = (trble_client_init_config *)(param->param[0]);
+		trble_server_init_config *server = (trble_server_init_config *)(param->param[1]);
+
+		ret = ble_drv_init(client, server);
+		if (ret != TRBLE_SUCCESS) {
+			BLE_LOG_ERROR("[BLEMGR] init fail[%d]\n", ret);
+			break;
+		}
+		g_state_handle.state = BLEMGR_INITIALIZED;
+#if defined(CONFIG_AMEBAD_BLE_SCATTERNET)
+		g_state_handle.mode = BLEMGR_COMBO_MODE;
+#elif defined(CONFIG_AMEBAD_BLE_CENTRAL)
+		g_state_handle.mode = BLEMGR_CLIENT_MODE;
+#elif defined(CONFIG_AMEBAD_BLE_PERIPHERAL)
+		g_state_handle.mode = BLEMGR_SERVER_MODE;
+#endif
+	} break;
+
+	case BLE_EVT_DEINIT_CMD: {
+		BLE_STATE_CHECK;
+
+		ret = ble_drv_deinit();
+		if (ret != TRBLE_SUCCESS) {
+			BLE_LOG_ERROR("[BLEMGR] deinit fail[%d]\n", ret);
+			break;
+		}
+		g_state_handle.state = BLEMGR_UNINITIALIZED;
+		g_state_handle.mode = BLEMGR_UNKNOWN_MODE;
+	} break;
+
+	case BLE_EVT_GET_MAC_CMD: {
+		BLE_STATE_CHECK;
+
+		uint8_t *mac = (uint8_t *)msg->param;
+		ret = ble_drv_get_mac_addr(mac);
+	} break;
+	case BLE_EVT_DISCONNECT_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_conn_handle con_handle = *(trble_conn_handle *)param->param[0];
+		trble_mode_e mode = *(trble_mode_e *)param->param[1];
+
+		ret = ble_drv_disconnect(con_handle, mode);
+	} break;
+
+	// Client
+	case BLE_EVT_DEL_BOND_CMD: {
```

Comment:
There are no `break` from here. Then whole codes will run.
Please check!!!

Reply:
Ok, this is critical issue. I will fix it